design/one-pager-resource-status-improvements.md

```diff
@@ -0,0 +1,91 @@
+
+To paraphrase, the conditions should, to the best of their ability, give the user and other components a precise status of the resource. A resource's conditions are also part of it's API, so changes should be well thought through.
+
+These proposals try to make the API for resources managed by crossplane providers to give users the best possible status about the health of their resources. 
+
+### 1: Improve Clarity of `Ready` Condition
+#### Problem
+If a user updates a managed resource in Crossplane today, there is no condition indicating this. The resource remains `Available` and no attempt is made at immediately observing the resource to check if it matches the desired state. Even if the update fails, the resource will be displayed as `Available`. 
+This means that a user cannot trust that an `Available` resource is a compliant resource, which can cause misinterpretation of the resource's state.
+
+Here is a diagram showing how the `Ready` condition is updated throughout the reconciler:
+##### Current `Ready` Condition Update Behavior
+```mermaid
+graph LR
+    Start(Reconcile Start)
+
+    NeedsDelete{Needs deleting?}
+    NeedsCreate{Needs creation?}
+    UpToDate{Up to date?}
+
+    Delete(Deletes the resource. \nReconciles immediately to make sure it is deleted. \nSets Ready: \nStatus = False \nReason = Deleting)
+    Create(Creates the resource. \nReconciled immediately to make sure it is created. \nSets Ready: \nStatus = False \nReason = Creating)
+    Observe(Doesn't need to do anything and doesn't change Ready. \nProvider author has to manually set Ready)
+    Update(Updates the resource. \nDoesn't reconcile immediately. \nDoesn't change ready status, so it remains `Ready = True`)
+
+    Start --> NeedsDelete
+    NeedsDelete -->|Yes| Delete
+    NeedsDelete -->|No| NeedsCreate
+    NeedsCreate -->|Yes| Create
+    NeedsCreate -->|No| UpToDate
+    UpToDate -->|Yes| Observe
+    UpToDate -->|No| Update
+```
+
+#### Solution
+Implement an `Updating` status, mimicking the `Creating` status. Setting this status every time an update request either succeeds or fails, informing a user that an update request has been requested, but we have not verified if the resource is compliant with the updated desired state.
+
+This also requires updating the reconciler to immediately requeue a reconcile after a successfuly requested update. This ensures that an updated resource is immediatly checked for compliance, to give user quick feedback on if the resource is `Available` again after the update.
+
+Note that this would technically change the definition of the `Ready` condition. However, this definition is not very well documented, so I am not sure how many know about it.  
```

Comment:
I do not think we should change the semantics or meaning of our long standing `Ready` condition. Way too many folks have made (codified) assumptions about the meaning across the ecosystem, that it's not an option IMO to change its meaning.  As you already mentioned above from the [API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties): 

> Once defined, the meaning of a Condition can not be changed arbitrarily - it becomes part of the API, and has the same backwards- and forwards-compatibility concerns of any other part of the API.

Reply:
I totally get what you mean! As you pointed out, I commented on this myself, so it was always on my mind. 

You know much better than me what others in the community have created, so if you think this is too much of a breaking change, I can understand that. 
But do you think there are that many that have created things with a strong assumption that `Ready` is not about desired state matching observed state? My point is just that, while I agree this would be a breaking change, I think it might be worth looking into if it would actually break anything.

Reply:
I'm sure people have yes - unfortunately it's impossible to ever know that fully within our ecosystem and it's many thousands of adopters. We can never hear from everyone and therefore we always have to stick with the fundamental principle of no breaking changes or changes to semantics in anything that is v1 API.